### PR TITLE
[DomCrawler] deprecate parents() in favor of ancestors()

### LINKED ISSUE #26432: [DomCrawler] Inconsistency between children and parents
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | all

According to the documentation:

> Get all the child or parent nodes:
>
> $crawler->filter('body')->children();
> $crawler->filter('body > p')->parents();

However, `children()` returns only direct children (not grandchildren...), while `parents()` returns all ancestors (including grandparents...).

Maybe it's more a docs issue than a code issue? Anyway it looks weird to me that methods with similar names have a different behavior.

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | yes
| Tickets       | Fix #26432
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

**Review by dunglas — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -494,13 +494,27 @@ public function previousAll()
...
-     * Returns the parents nodes of the current selection.
+     * Returns the parent nodes of the current selection.
      *
      * @return static
      *
      * @throws \InvalidArgumentException When current node is empty
      */
     public function parents()
+    {
+        @trigger_deprecation('symfony/dom-crawler', '5.3', sprintf('The %s() method is deprecated, use ancestors() instead.', __METHOD__));
```
Comment (ro0NL) [inline src/Symfony/Component/DomCrawler/Crawler.php:505]: AFAIK the `@` isnt needed

Comment (chalasr) [reply src/Symfony/Component/DomCrawler/Crawler.php:505]: fixed in https://github.com/symfony/symfony/commit/e970027682e27b2605996e6ae3c082dde2a8d828
good catch, thank you!

Comment (xabbuh) [reply src/Symfony/Component/DomCrawler/Crawler.php:505]: Thank you @chalasr!

**Pull request merged**